src/algebra/algebra/basic.lean

```diff
@@ -1538,3 +1538,15 @@ begin
 end
 
 end submodule
+
+section non_associative
+
+variables {R A : Type*}
+variables [semiring R] [non_unital_non_assoc_semiring A] [module R A]
+variables [is_scalar_tower R A A] [smul_comm_class R A A]
+
+lemma algebra.smul_mul_smul
+  (r s : R) (x y : A) : (r • x) * (s • y) = (r * s) • (x * y) :=
+smul_smul_smul_comm r s x y
```

Comment:
How is this different to the existing `smul_mul_smul`?

Reply:
The statement is the same but its hypotheses are different.

Reply:
Can you show the output of `#check` here for both?

Reply:
Of course, I should have done that before.
```lean
smul_mul_smul :
  ∀ {M : Type u_3} {α : Type u_4} [_inst_1 : monoid M] [_inst_2 : mul_action M α] [_inst_3 : monoid α] (r s : M)
  (x y : α) [_inst_4 : is_scalar_tower M α α] [_inst_5 : smul_comm_class M α α],
    r • x * s • y = (r * s) • (x * y)
```
versus
```lean
algebra.smul_mul_smul :
  ∀ {R : Type u_3} {A : Type u_4} [_inst_1 : semiring R] [_inst_2 : non_unital_non_assoc_semiring A]
  [_inst_3 : module R A] [_inst_4 : is_scalar_tower R A A] [_inst_5 : smul_comm_class R A A] (r s : R) (x y : A),
    r • x * s • y = (r * s) • (x * y)
```

Reply:
I had persuaded myself it was OK to have both but I can see the argument that it's worth working to unify them.